src/validation.cpp

```diff
@@ -1614,6 +1614,7 @@ void static FlushBlockFile(bool fFinalize = false)
         if (fFinalize)
             TruncateFile(fileOld, vinfoBlockFile[nLastBlockFile].nSize);
         FileCommit(fileOld);
+        DirectoryCommit(GetDataDir() / "blocks");
```

Comment:
Realize you are currently working on this, but if you decide to go with this code as is, you should definitely comment here that call this isn't needed in the case where new files aren't being added, in case somebody wants to try to optimize this later.

Reply:
Won't it be a reasonably fast noop if the directory hasn't changed?